slsim/lens.py

```diff
@@ -341,41 +311,53 @@ def external_shear(self):
         """
         gamma1, gamma2, _ = self.los_linear_distortions
         return (gamma1**2 + gamma2**2) ** 0.5
-
+    
     @property
     def einstein_radius_deflector(self):
         """Einstein radius, from SIS approximation (coming from velocity dispersion)
         without line-of-sight correction.
 
-        :return:
-        """
-        if not hasattr(self, "_theta_E"):
-            if self.deflector.redshift >= self.source.redshift:
-                self._theta_E = 0
-            elif self.deflector.deflector_type in ["EPL"]:
-                self._theta_E = self._lens_cosmo.sis_sigma_v2theta_E(
-                    float(self.deflector.velocity_dispersion(cosmo=self.cosmo))
-                )
-            else:
-                # numerical solution for the Einstein radius
-                lens_model_list, kwargs_lens = self.deflector_mass_model_lenstronomy()
-                lens_model = LensModel(lens_model_list=lens_model_list)
-                lens_analysis = LensProfileAnalysis(lens_model=lens_model)
-                self._theta_E = lens_analysis.effective_einstein_radius(
-                    kwargs_lens, r_min=1e-3, r_max=5e1, num_points=50
-                )
-        return self._theta_E
+        :return: list of einstein radius of each lens-source pair.
+        """
+        if not hasattr(self, "_theta_E_list"):
+            self._theta_E_list = []
+            for source in self.source:
+                if self.deflector.redshift >= source.redshift:
+                    self._theta_E_list.append(0)
+                elif self.deflector.deflector_type in ["EPL"]:
+                    _lens_cosmo = LensCosmo(
+                                    z_lens=float(self.deflector.redshift),
+                                    z_source=float(source.redshift),
+                                    cosmo=self.cosmo,
+                                )
+                    _theta_E = _lens_cosmo.sis_sigma_v2theta_E(
+                        float(self.deflector.velocity_dispersion(cosmo=self.cosmo))
+                    )
+                    self._theta_E_list.append(_theta_E)
+                else:
+                    # numerical solution for the Einstein radius
+                    lens_model_list, kwargs_lens = self.deflector_mass_model_lenstronomy()
+                    lens_model = LensModel(lens_model_list=lens_model_list)
```

Comment:
for different source redshifts, this does not work. This way it only works for z_source=z_source_convention. I'll update the lenstronomy version such that we can adopt to it